app/models/chargeback_rate_detail.rb

```diff
@@ -1,19 +1,44 @@
 class ChargebackRateDetail < ActiveRecord::Base
   belongs_to :chargeback_rate
   belongs_to :detail_measure, :class_name => "ChargebackRateDetailMeasure", :foreign_key => :chargeback_rate_detail_measure_id
-
-  validates :rate, :numericality => true
+  has_many :chargeback_tiers, :dependent => :destroy
   validates :group, :source, :presence => true
+  validate :complete_tiers
+
+  # Set the rates according to the tiers
+  def rate(value)
+    @fix_rate = 0.0
+    @var_rate = 0.0
+    ChargebackTier.where(:chargeback_rate_detail_id => id).each do |tier|
```

Comment:
This should use the Active Record association instead. Like this:
`chargeback_tiers .each do |tier|`